opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/BaseInteractor.java

```diff
@@ -191,41 +204,45 @@ public void run() {
                     structure.setProperties(properties);
                     structure.setSyncStatus(BaseRepository.TYPE_Created);
                     structureRepository.addOrUpdate(structure);
-
-                    Task task = new Task();
-                    task.setIdentifier(UUID.randomUUID().toString());
-                    task.setCampaignIdentifier(PreferencesUtil.getInstance().getCurrentCampaignId());
-                    task.setGroupIdentifier(operationalAreaId);
-                    task.setStatus(Task.TaskStatus.READY);
-                    task.setBusinessStatus(BusinessStatus.NOT_VISITED);
-                    task.setPriority(3);
                     Context applicationContext = RevealApplication.getInstance().getApplicationContext();
-                    if (StructureType.RESIDENTIAL.equals(structureType)) {
-                        task.setCode(Intervention.IRS);
-                        task.setDescription(applicationContext.getString(R.string.irs_task_description));
-                        task.setFocus(Intervention.IRS_VISIT);
-                    } else if (StructureType.MOSQUITO_COLLECTION_POINT.equals(structureType)) {
-                        task.setCode(Intervention.MOSQUITO_COLLECTION);
-                        task.setDescription(applicationContext.getString(R.string.mosquito_collection_task_description));
-                        task.setFocus(Intervention.MOSQUITO_COLLECTION);
-                    } else if (StructureType.LARVAL_BREEDING_SITE.equals(structureType)) {
-                        task.setCode(Intervention.LARVAL_DIPPING);
-                        task.setDescription(applicationContext.getString(R.string.larval_dipping_task_description));
-                        task.setFocus(Constants.Intervention.LARVAL_DIPPING);
+                    Task task;
+                    if (StructureType.RESIDENTIAL.equals(structureType) && Utils.getInterventionLabel() == R.string.focus_investigation) {
+                        task = taskUtils.generateRegisterFamilyTask(applicationContext, structure.getId());
+                    } else {
+                        task = new Task();
+                        task.setIdentifier(UUID.randomUUID().toString());
+                        task.setCampaignIdentifier(PreferencesUtil.getInstance().getCurrentCampaignId());
```

Comment:
A lot of these property setting is already done in the method `generateRegisterFamilyTask`.

Can these be further factored out into a method?

Reply:
Done.
I have called the `org.smartregister.reveal.util.TaskUtils#generateTask` to generate the tasks